backend/src/services/authManager/providers/discord.ts

```diff
@@ -0,0 +1,33 @@
+import { OAuthUser } from "../../../models/users";
+
+interface DiscordUser {
+  id: string;
+}
+
+const getUserFromAccessToken = async (
```

Comment:
Why you implemented your own fetch profile function when next auth has one build-in in the discord provider?

Reply:
Does it? The only thing that I can see is the Discord provider that one of the fields of it's the `userinfo` that means the endpoint from which the user info is extracted.

Reply:
Oh I see, I see these two cases as two different things:

What I'm doing in my code is checking that the `accessToken` that the user has sent is a correct discord user. So my input is an `accessToken` and the output is Discord's interface for Users. 

Meanwhile, the profile function it's a callback on how to treat the user interface (for mapping it to the desired user DTO the app needs, ensure certain consistency on the backend, etc.). Its input is the already retrieved User rather than the `accessToken`. 

Also, may be important to note is that the backend does not use next-auth it's only used in the frontend.

Reply:
You are right, I had not realized this file was in backend part 🙈, sorry 👍